subworkflows/local/vcf_normalization/main.nf

```diff
@@ -0,0 +1,50 @@
+// Normalize all unannotated VCFs
+
+// Import modules
+include { ADD_INFO_TO_VCF  } from '../../../modules/local/add_info_to_vcf/main'
+include { TABIX_BGZIPTABIX as TABIX_EXT_VCF } from '../../../modules/nf-core/tabix/bgziptabix/main'
+include { BCFTOOLS_NORM as VCFS_NORM } from '../../../modules/nf-core/bcftools/norm/main'
+include { BCFTOOLS_SORT as VCFS_NORM_SORT } from '../../../modules/nf-core/bcftools/sort/main'
+include { TABIX_TABIX as TABIX_VCFS_NORM_SORT } from '../../../modules/nf-core/tabix/tabix/main'
+
+// Workflow to normalize, compress, and index VCF files
+workflow NORMALIZE_VCFS {
+
+    take:
+    germline_vcfs
+    tumor_only_vcfs
+    somatic_vcfs
+    fasta
+
+    main:
+    versions = Channel.empty()
+
+    vcfs = germline_vcfs.mix(tumor_only_vcfs, somatic_vcfs)
+
+    // Add additional information to VCF files
+    ADD_INFO_TO_VCF(vcfs)
+
+    // Compress the VCF files with bgzip
+    TABIX_EXT_VCF(ADD_INFO_TO_VCF.out.vcf)
+
+    // Normalize the VCF files with BCFTOOLS_NORM
+    VCFS_NORM(TABIX_EXT_VCF.out.gz_tbi, fasta)
+
+    // Sort the normalized VCF files
+    VCFS_NORM_SORT(VCFS_NORM.out.vcf)
+
+    // Index the sorted normalized VCF files
+    TABIX_VCFS_NORM_SORT(VCFS_NORM_SORT.out.vcf)
+
+    // Gather versions of all tools used
+    versions = versions.mix(ADD_INFO_TO_VCF.out.versions)
+    versions = versions.mix(VCFS_NORM.out.versions)
+    versions = versions.mix(TABIX_EXT_VCF.out.versions)
+    versions = versions.mix(VCFS_NORM_SORT.out.versions)
+    versions = versions.mix(TABIX_VCFS_NORM_SORT.out.versions)
+
+    emit:
+    vcfs = VCFS_NORM_SORT.out.vcf // normalized vcfs
```

Comment:
I think that one should do the trick, but we need to figure out what to do with the tbis, and what's happening with the tbis on the concatenate side

Reply:
Now, when I run tests for concatenate_vcfs and normalize_vcfs separately, they complete without any errors. The resulted ouputs are vcf.gz as well as vcf.gz.tbi (variant_calling/normalized/testN/testN.norm.vcf.gz, testN.norm.vcf.gz.tbi, testN.vcf.gz and variant_calling/concat/testN/testN.germline.vcf.gz, testN.germline.vcf.gz.tbi, ),. In case of normalized, there is an extra vcf.gz file (testN.vcf.gz) in the outdir, which I don't know where it comes from and if it causes any issues.
But in case I run test for both concatenate and normalize, the test fails with following warnings and errors - I think, the same we've encountered during the monday's meeting with Maxime.
![Snímek obrazovky 2024-11-27 v 10 39 48 (2)](https://github.com/user-attachments/assets/2a271239-8d4a-470f-a11a-59a4276759d3)

Reply:
@maxulysse So I've tried, but local tests worked (see attached), but here both concat and norm are failing all over again ...no idea, why
![Uploading Snímek obrazovky 2024-11-28 v 13.00.53 (2).png…]()